### PR TITLE
feat: searchbar added

### PR DESCRIPTION
**Pull request by sakshamAK**
Now you can search your items according to your artist name, category name, and also product name.

start time: 0:08

https://user-images.githubusercontent.com/55906931/160270696-db7109f1-4487-496f-b849-cd27cd87d054.mp4

### CODE REVIEW

File: src/backend/db/products.js [hunk 1]
```diff
@@ -125,7 +125,8 @@ export const products = [
     onSale: false,
     starRatings: 5,
     mountType: "Wall",
-    color: "blue"
+    color: "blue",
+    qty: 1
```
Comment (SJTGSHIVAM) [inline src/backend/db/products.js:129]: Just a suggestion. you can always share the context in pr description if you have changed anything other than that is needed.

File: src/pages/Cart/Cart.jsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ export const Cart = () => {
                         <h3>Your Cart Is Empty</h3>
                     </div> 
                     
-                    : mycart?.map(({ _id, imgSrc, title, type, price, inStock, fastDelivery, onSale }) => (
+                    : mycart?.map(({ _id, imgSrc, title, type, price, discount, inStock, fastDelivery, onSale, qty }) => (
                         <Fragment>
```
Comment (SJTGSHIVAM) [inline src/pages/Cart/Cart.jsx:30]: <> works same as <Fragment>. I know this one is a little opinionated but you can pick anything from both.

File: src/pages/Cart/Cart.jsx [hunk 3]
```diff
@@ -39,9 +39,9 @@ export const Cart = () => {
                                 </div>
                             </div>
                             <div className={`${styles.updateItems}`}>
-                                <button>-</button>
-                                <input type="text" value="1" />
-                                <button>+</button>
+                                <button onClick = { () => dispatch(decreaseItem({ _id, imgSrc, title, type, price, inStock, fastDelivery, onSale, discount, qty })) }>-</button>
```
Comment (SJTGSHIVAM) [inline src/pages/Cart/Cart.jsx:42]: generally when we have to type a lot of values with " , "
Everything on a separate line makes a lot more sense and the cherry on the cake: its more readable as well.
```
return {
        id,
        fname,
        username,
        encodedToken,
        cartLength: Number(cartLength),
        wishlistLength: Number(wishlistLength),
      };

File: src/pages/ProductList/ProductList.jsx [hunk 4]
```diff
@@ -37,27 +22,12 @@ export const ProductList = () => {
...
-                                    <p className="prod-det">Rs.{ discount } <span>Rs.{ price } </span></p>
-                                    {fastDelivery && <p className = {`${styles.itemAvailable} prod-cate`}>FAST DELIVERY</p>}
-                                    {onSale && <p className = {`${styles.itemAvailable} prod-cate`}>ON SALE</p>}
-                                    <div className="btn primary" onClick = { () => cartState.mycart.find(item => item._id === _id) ? (1 === 2) ? navigate("/cart") : navigate("/signin") : dispatch(addToCart({ _id, imgSrc, title, type, price, inStock, fastDelivery, onSale, discount })) }>Buy Now</div>
-                                </div>
-                            </div>
+                        {filteredData(productState, products)?.map(({_id, imgSrc, title, type, price, inStock, fastDelivery, onSale, discount, qty}) => (
+                        <ProductListComponent _id = {_id} imgSrc = {imgSrc} title = {title} type = {type} price = {price} inStock = {inStock} fastDelivery = {fastDelivery} onSale = {onSale} discount = {discount} qty = {qty} />
```
Comment (SJTGSHIVAM) [inline src/pages/ProductList/ProductList.jsx:42]: Same multiline pattern is useful here as well for eg:
```
         <input
            type="text"
            placeholder="Search"
            className="tui__input--val"
            id="search"
          />

**Review by SJTGSHIVAM — APPROVED** [review]
Overall UI is clean. A great job is done by you there.

File: src/backend/db/products.js (hunk 1 above)
Comment (sakshamAK) [reply src/backend/db/products.js:129]: oops forgot to update, Thanks!

File: src/pages/Cart/Cart.jsx (hunk 3 above)
Comment (sakshamAK) [reply src/pages/Cart/Cart.jsx:42]: thanks

File: src/pages/ProductList/ProductList.jsx (hunk 4 above)
Comment (sakshamAK) [reply src/pages/ProductList/ProductList.jsx:42]: sure Thanks

**Review by mimansha-swarup — APPROVED** [review]
remove unrequired comments
and use prettier format file once before pushing to github code will be more redable
alt+shift+f shortcut for formatting

**Pull request merged**